packages/aws-cdk/lib/context-providers/vpcs.ts

```diff
@@ -168,33 +159,37 @@ function getTag(name: string, tags?: AWS.EC2.Tag[]): string | undefined {
 /**
  * Group subnets of the same type together, and order by AZ
  */
-function groupSubnets(subnets: Subnet[]): SubnetGroups {
-  const grouping: {[key: string]: Subnet[]} = {};
+function groupSubnets(subnets: Subnet[]): cxapi.VpcSubnetGroup[] {
+  const grouping: { [key: string]: Subnet[] } = {};
   for (const subnet of subnets) {
     const key = [subnet.type, subnet.name].toString();
-    if (!(key in grouping)) { grouping[key] = []; }
+    if (!(key in grouping)) {
+      grouping[key] = [];
+    }
     grouping[key].push(subnet);
   }
 
-  const groups = Object.values(grouping).map(sns => {
-    sns.sort((a: Subnet, b: Subnet) => a.az.localeCompare(b.az));
+  return Object.values(grouping).map(subnetArray => {
+    subnetArray.sort((subnet1: Subnet, subnet2: Subnet) => subnet1.az.localeCompare(subnet2.az));
+
     return {
-      type: sns[0].type,
-      name: sns[0].name,
-      subnets: sns,
+      name: subnetArray[0].name,
+      type: subnetTypeToVpcSubnetType(subnetArray[0].type),
+      subnets: subnetArray.map(subnet => ({
+        subnetId: subnet.subnetId,
+        availabilityZone: subnet.az,
+        routeTableId: subnet.routeTableId,
+      })),
     };
   });
+}
 
-  const azs = groups[0].subnets.map(s => s.az);
-
-  for (const group of groups) {
-    const groupAZs = group.subnets.map(s => s.az);
-    if (!arraysEqual(groupAZs, azs)) {
-      throw new Error(`Not all subnets in VPC have the same AZs: ${groupAZs} vs ${azs}`);
-    }
+function subnetTypeToVpcSubnetType(type: SubnetType): cxapi.VpcSubnetGroupType {
```

Comment:
Oh do you suppose the lack of another level mapping was the cause of broken subnet type detection? What is this mapping to what?

Reply:
Nah. This is just a simple translation between 2 enums (`SubnetType` is defined in `@aws-cdk/aws-ec2`, while `VpcSubnetGroupType` is in `@aws-cdk/cx-api`.